src/Compiler/Symbols/Symbols.fs

```diff
@@ -430,18 +430,27 @@ type FSharpEntity(cenv: SymbolEnv, entity: EntityRef) =
             Some (buildAccessPath (Some cp))
         | Some _ -> None
 
-    member x.QualifiedName = 
+    member x.CompiledRepresentation =
         checkIsResolved()
-        let fail() = invalidOp (sprintf "the type '%s' does not have a qualified name" x.LogicalName)
-#if !NO_TYPEPROVIDERS
-        if entity.IsTypeAbbrev || entity.IsProvidedErasedTycon || entity.IsNamespace then fail()
-        #else
-        if entity.IsTypeAbbrev || entity.IsNamespace then fail()
+
+        let fail () =
+            invalidOp (sprintf $"the type '{x.LogicalName}' does not have a qualified name")
+
+#if !NO_EXTENSIONTYPING
```

Comment:
This was replaced by `NO_TYPEPROVIDERS` some time ago, I believe.

Reply:
Updated, thanks! This was cherry-picked from an older branch.